packages/vsx-registry/src/browser/vsx-extensions-contribution.ts

```diff
@@ -107,4 +172,22 @@ export class VSXExtensionsContribution extends AbstractViewContribution<VSXExten
         }
         return false;
     }
+
+    /**
+     * Installs a local .vsix file after prompting the `Open File` dialog. Resolves to the URI of the file.
+     */
+    protected async installFromVSIX(): Promise<URI | undefined> {
+        const props: OpenFileDialogProps = {
+            title: VSXExtensionsCommands.INSTALL_FROM_VSIX.dialogLabel,
+            openLabel: 'Install',
+            filters: { 'VSIX Extensions (*.vsix)': ['vsix'] }
+        };
+        const [rootStat] = await this.workspaceService.roots;
+        const destinationFileUri = await this.fileDialogService.showOpenDialog(props, rootStat);
+        if (destinationFileUri) {
+            await this.pluginServer.deploy(`local-file:${destinationFileUri.path}`);
+            return destinationFileUri;
```

Comment:
Instead of duplicating the code, should we not call the method directly? (through the command registry)